### ISSUE TITLE
Refactor CICD script to decouple with travis

### ISSUE THREAD

**Issue by SamYuan1990** [Hyperledger-TWGC/Gm-Go#5]
**Is your feature request related to a problem? Please describe.**
Currently we are on travis for CI however fabric itself on azure.

**Describe the solution you'd like**
Plan A:
Following https://docs.microsoft.com/zh-cn/azure/devops/pipelines/migrate/from-travis?view=azure-devops
starts migration when we need.

Plan B:
Travis as env, which only prepare for VM.
In this repo having a `CI.sh $HOME $TRAVIS_BUILD_DIR` (for sample) and making Travis invokes this shell.
When we migration, just making Azure pipeline invokes `CI.sh`

CI.sh 
```
apt-get update && apt-get install xxxx
cd $1
git clone https://github.com/guanzhi/GmSSL.git
cd GmSSL
./config
make
make install_sw
export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
sudo ldconfig
echo completed for xxx
echo cat $PATH // or the libraray/ .so files changed in above scripts.
cd $2
cd gmssl
go build
echo test begin // for this I am not sure should using go invoke CLI command curl --version ?
curl --version
cd test
go test
```

**Describe alternatives you've considered**
- Considering if we may move to azure from travis, we'd better refactor our scripts.
- Considering Azure env may diff with travis env.
- Considering if we need further debugging, see currently we just use `curl`, when a line with fabric, we may need to know which .so file changed, as it possible may impact others we don't know so far.

**Additional context**

**Comment by davidkhala** [comment]
My suggestion is not to migrate or decouple, but keep two CI in parallel. 
Meanwhile we could configure AZP as default checker. 

The curl test I added here is a smoke test that reflecting current environment pollution problem. #3

**Comment by davidkhala** [comment]
@zhaoxiaomeng After digging into the root cause of why `make` fails, I find that `./config` could not recognize AZP agent VM OS properly:

On my desktop it is recognized as `Operating system: x86_64-whatever-linux2`
But in AZP Ubuntu 18 it shows [Operating system: x86_64-whatever-build](https://dev.azure.com/Hyperledger/TWGC-Fabric/_build/results?buildId=17742&view=logs&j=eb5ce242-5842-55b4-e135-32b31d565442&t=eecfea67-78e0-5123-2ba5-8df5e83c46e3&l=14)

**Comment by SamYuan1990** [comment]
@zhaoxiaomeng , @davidkhala 

not sure information below may help?
https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml
https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md

**Comment by SamYuan1990** [comment]
ok... just closed #7 and curious on how to get it fixed? @zhaoxiaomeng
